operators/src/processing/bandwise_expression/mod.rs

```diff
@@ -0,0 +1,438 @@
+use geoengine_datatypes::primitives::RasterQueryRectangle;
+use geoengine_datatypes::raster::{
+    GridOrEmpty2D, MapElementsParallel, Pixel, RasterDataType, RasterTile2D,
+};
+use geoengine_expression::{
+    DataType, ExpressionAst, ExpressionParser, LinkedExpression, Parameter,
+};
+use serde::{Deserialize, Serialize};
+
+use super::expression::get_expression_dependencies;
+use super::RasterExpressionError;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BandwiseExpressionParams {
+    pub expression: String,
+    pub output_type: RasterDataType,
+    pub map_no_data: bool,
+    // TODO: new unit for each band?
+}
+
+/// This `QueryProcessor` performs a unary expression on all bands of its input raster series.
+pub type BandwiseExpression = Operator<BandwiseExpressionParams, SingleRasterSource>;
+
+impl OperatorName for BandwiseExpression {
+    const TYPE_NAME: &'static str = "BandwiseExpression";
+}
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for BandwiseExpression {
+    async fn _initialize(
+        self: Box<Self>,
+        path: WorkflowOperatorPath,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let name = CanonicOperatorName::from(&self);
+
+        let source = self.sources.initialize_sources(path, context).await?.raster;
+
+        let in_descriptor = source.result_descriptor();
+
+        // TODO: ensure all bands have same measurement unit?
+
+        let result_descriptor = in_descriptor.map_data_type(|_| self.params.output_type);
+
+        let parameters = vec![Parameter::Number("x".into())];
+
+        let expression = ExpressionParser::new(&parameters, DataType::Number)
+            .map_err(RasterExpressionError::from)?
+            .parse(
+                "expression", // TODO: what is the name used for?
+                &self.params.expression,
+            )
+            .map_err(RasterExpressionError::from)?;
+
+        Ok(Box::new(InitializedBandwiseExpression {
+            name,
+            result_descriptor,
+            source,
+            expression,
+            map_no_data: self.params.map_no_data,
+        }))
+    }
+
+    span_fn!(BandwiseExpression);
+}
+
+pub struct InitializedBandwiseExpression {
+    name: CanonicOperatorName,
+    result_descriptor: RasterResultDescriptor,
+    source: Box<dyn InitializedRasterOperator>,
+    expression: ExpressionAst,
+    map_no_data: bool,
+}
+
+impl InitializedRasterOperator for InitializedBandwiseExpression {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let typed_raster_processor = self.source.query_processor()?.into_f64();
+
+        let output_type = self.result_descriptor().data_type;
+
+        // TODO: spawn a blocking task for the compilation process
```

Comment:
could you do that?

Reply:
no, because the `query_processor` function is not async. I just copied the code here from the other expression operator inclusive the todo